x-pack/plugins/security_solution/server/endpoint/migrations/turn_off_agent_policy_features.test.ts

```diff
@@ -23,12 +25,16 @@ describe('Turn Off Agent Policy Features Migration', () => {
     turnOffAgentPolicyFeatures(fleetServices, productFeatureService, logger);
 
   beforeEach(() => {
-    const endpointContextStartContract = createMockEndpointAppContextServiceStartContract();
+    const mockedFleetServices = createEndpointFleetServicesFactoryMock();
 
-    ({ logger } = endpointContextStartContract);
-
-    productFeatureService = endpointContextStartContract.productFeaturesService;
-    fleetServices = endpointContextStartContract.endpointFleetServicesFactory.asInternalUser();
+    fleetServices = mockedFleetServices.service.asInternalUser();
+    logger = loggingSystemMock.createLogger();
+    productFeatureService = createProductFeaturesServiceMock(
```

Comment:
do you think we could refactor `createProductFeaturesServiceMock` to accept an object, instead of multiple arguments where some of them are passed as `undefined` ?

Reply:
I'm ok either way... but perhaps if you feel we should, lets address it in a separate PR